scipy/spatial/transform/_rotation.pyx

```diff
@@ -464,6 +464,10 @@ cdef inline double[:, :] _compose_quat(
 
     return product
 
+def compose_quat(p, q):
```

Comment:
If we use `cdef` and `cimport`, will it be more efficient?

Reply:
I tried this before, but it failed with

```
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      from scipy._lib._util import check_random_state, _transition_to_rng
      from ._rotation import Rotation
      
      cimport numpy as np
      cimport cython
      from ._rotation cimport compose_quat
      ^
      ------------------------------------------------------------
      
      /home/afabisch/Projekte/scipy/scipy/spatial/transform/_rigid_transformation.pyx:9:0: 'scipy/spatial/transform/_rotation.pxd' not found
      ninja: build stopped: subcommand failed.
```

I'm not a Cython expert. Do we want to add a .pxd file? Do we have to configure something? Is there another way?

Reply:
I guess the simplest way is to create something like `_common.pxd` and use `inline` implementation of `compose_quat` here. I'll try to implement that.

Reply:
Done in a PR into this branch.